internal/server/data/identity.go

```diff
@@ -34,6 +34,23 @@ func (i *identitiesTable) ScanFields() []any {
 	return []any{&i.CreatedAt, &i.CreatedBy, &i.DeletedAt, &i.ID, &i.LastSeenAt, &i.Name, &i.OrganizationID, &i.SSHLoginName, &i.UpdatedAt, &i.VerificationToken, &i.Verified}
 }
 
+// diff compares two string slices (x and y), it returns a slice of strings that exist in x that do not exist in y
+func diff(x, y []string) []string {
+	// this modifies the order of x and y, but its ok in this case since the order of the groups assigned to a user does not matter
+	slices.Sort(x)
+	slices.Sort(y)
+
+	difference := []string{}
+
+	for _, val := range x {
+		if !slices.Contains(y, val) {
```

Comment:
I believe that since we're using `Contains` on `y` that `y` doesn't need to be sorted.

I think an implementation that sorts both and traverses both with indexes might be slightly more efficient than using `Contains`, but I doubt it's worth the optimization at this point.